v2/stream/new/subscription.go

```diff
@@ -0,0 +1,142 @@
+package new
+
+import (
+	"errors"
+
+	"github.com/vmihailenco/msgpack/v5"
+)
+
+var ErrSubChangeBeforeConnect = errors.New("subscription change attempted before connecting")
+var ErrSubChangeAfterTerminated = errors.New("subscription change after client termination")
+var ErrSubChangeAlreadyInProgress = errors.New("subscription change already in progress")
+var ErrSubChangeInterrupted = errors.New("sub change interrupted by client termination")
```

Comment:
[minor] all above errors spell out "subscription". Either do the same or change them too. I prefer the former.